app/assets/javascripts/application.js

```diff
@@ -20,3 +20,13 @@ jQuery.ajaxSetup({
     xhr.setRequestHeader("Accept", "text/javascript");
   }
 });
+
+$("#search_form").keyup(delay(function() { $('form').find('input').submit()}, 500));
```

Comment:
Just noticed we include underscore.js already, so this could use https://underscorejs.org/#debounce rather than defining this delay function.

I found an example at https://jsfiddle.net/missinglink/19e2r2we/